emantomo/protocols/protocol_tomo_extraction_from_tomo.py

```diff
@@ -332,15 +333,15 @@ def getInputTomograms(self):
         else:
             return self.inputTomograms.get()
 
-    def readSetOfSubTomograms(self, tomoFile, outputSubTomogramsSet, coordSet, volId, factor):
+    def readSetOfSubTomograms(self, tomoFile, outputSubTomogramsSet, coordSet, volId, factor, counter):
 
         self.info("Registering subtomograms for %s" % tomoFile)
 
         outRegex = self._getExtraPath(pwutils.removeBaseExt(tomoFile) + '-*.mrc')
         subtomoFileList = sorted(glob.glob(outRegex))
 
         # ih = ImageHandler()
-        for counter, subtomoFile in enumerate(subtomoFileList):
+        for idx, subtomoFile in enumerate(subtomoFileList):
```

Comment:
I think this assumption is also dangerous. It assumes later in the code that coordinates id matches, this will probably break with subsets?

Reply:
Again, coordset made me think it was a setofcoordinates, but it isn't. I've tested extraction after subsets and joins and all went perfect.